pokemon_service/controllers/pokemon_controller.go

```diff
@@ -0,0 +1,79 @@
+package controller
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	repository "github.com/PasHdez/ondemand-go-bootcamp/repository"
+	"github.com/julienschmidt/httprouter"
+)
+
+// controller is the implementation of Controller interface
+type controller struct {
+	repository repository.PokemonRepository
+}
+
+// Controller is the interface that provides the controller methods
+type Controller interface {
+	GetPokemon(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+	GetPokemons(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+	GetPokemonsPars(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+}
+
+// NewController returns a new instance of Controller
+func NewPokemonController(repository repository.PokemonRepository) Controller {
+	return &controller{repository}
+}
+
+// GetPokemons returns a json response with all pokemons
+func (c *controller) GetPokemons(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+
+	poks, err := c.repository.GetPokemons()
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	json.NewEncoder(w).Encode(poks)
+}
+
+// GetPokemon returns a json response with a pokemon by id
+func (c *controller) GetPokemon(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	w.Header().Set("Content-Type", "application/json")
+	id := ps.ByName("id")
+	idInt, err := strconv.Atoi(id)
```

Comment:
you can save the assignation of `id`
```suggestion
	idInt, err := strconv.Atoi(ps.ByName("id"))
```